packages/common/src/services/audius-backend/AudiusBackend.ts

```diff
@@ -1683,15 +1686,27 @@ export const audiusBackend = ({
    * @params {string} ethAddress - Optional ETH wallet address to derive user bank. Defaults to hedgehog wallet
    * @returns {Promise<BN>} balance or null if failed to fetch balance
    */
-  async function getWAudioBalance(ethAddress?: string): Promise<BN | null> {
-    await waitForLibsInit()
-
+  async function getWAudioBalance({
+    ethAddress,
+    sdk
+  }: {
+    ethAddress?: string
+    sdk: AudiusSdk
+  }): Promise<BN | null> {
     try {
-      const userBank = await audiusLibs.solanaWeb3Manager.deriveUserBank({
-        ethAddress
-      })
-      const ownerWAudioBalance =
-        await audiusLibs.solanaWeb3Manager.getWAudioBalance(userBank)
+      const { userBank } =
+        await sdk.services.claimableTokensClient.getOrCreateUserBank({
+          ethWallet: ethAddress ?? (await sdk.services.auth.getAddress()),
```

Comment:
I would just make `ethAddress` required and remove this fallback reading of the address entirely.

And I think we don't want to use the signing wallet address because in manager mode that would be the manager's address.
So you can either select the current user wallet address from the account slice, or you can do something like:

```ts
// This is in the function that is calling getWAudioBalance
const authService = yield* getContext('authService')
const { accountWalletAddress: ethAddress } = yield* call([authService, authService.getWalletAddresses])
if (!accountWalletAddress) {
  // error handling and bail early without calling getWAudioBalance
}

const balance = yield* call([audiusBackendInstance, audiusBackendInstance.getWaudioBalance], { sdk, ethAddress})
```

Reply:
ah duh just require it